### ISSUE TITLE
After the screen is locked, the background program stops running?

### ISSUE THREAD

**Issue by RNHaoHao** [ocetnik/react-native-background-timer#198]
After the screen is locked, the background program stops running?

**Comment by plastic** [comment]
I have the same problem

**Comment by RNHaoHao** [comment]
Is it that the plug-in itself has problems? Or is there something wrong with what we quoted?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Andre Andrade"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月6日(星期五) 晚上8:35
收件人:&nbsp;"ocetnik/react-native-background-timer"<react-native-background-timer@noreply.github.com&gt;;
抄送:&nbsp;"IT々じ浩Ю"<137717070@QQ.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [ocetnik/react-native-background-timer] After the screen is locked, the background program stops running? (#198)




I have the same problem

**Comment by EhsanSarshar** [comment]
the problem is with async call inside callback

**Comment by RNHaoHao** [comment]
Have you solved the problem? @EhsanSarshar

**Comment by teamhide** [comment]
Any solution here?

**Comment by EhsanSarshar** [comment]
@RNHaoHao there are plenty of solutions.
for my case I just need to send geolocation in background that solved with react-native-background-geolocation. 
you can use react-native-background-action or create your own implementation in native side.
or use expo inside your bare react native which is also possible and it's an awesome choice in most cases. expo have good modules for handling background task that now you can use within your react native project easily

**Comment by theagilecoder** [comment]
> After the screen is locked, the background program stops running?

Yes, background time works when app is minimized, but doesn't work when screen is lcoked.

**Comment by juanamd** [comment]
Hi,
I had this same issue and found no easy solution with this library, so I made my own which works even if the device is locked (among other improvements).
Only Android is supported, but you can still use this (or another) library for iOS in conjunction.

https://www.npmjs.com/package/react-native-background-timer-android

I've been using it in my production apps for more than a week with no issues.
